File: Insert Menu.php
<?php
/*
 * PSPEC:
 * File Name: insertMenu.php
 * Purpose: Menambahkan menu baru ke database.
 * Inputs: Nama menu, harga dari form.
 * Outputs: Menyimpan ke database tabel 'menu'.
 * Constraints: Nama menu dan harga tidak boleh kosong.
 *
 * DPPL:
 * 1. Terima input dari form POST.
 * 2. Validasi input.
 * 3. Simpan ke tabel 'menu'.
 */
include 'functions.php';

if ($_POST) {
    $menu_name = $_POST['menu_name'];
    $price = $_POST['price'];
    if ($menu_name && $price) {
        insertMenu($menu_name, $price);
        echo "Menu added!";
    } else {
        echo "All fields are required!";
    }
}
?>

File: Insert Menu.pas
{
  PSPEC:
  File Name: insert_menu.pas
  Purpose: Menambahkan menu baru ke sistem.
  Inputs: Nama menu, harga.
  Outputs: Menyimpan ke array menu.
  Constraints: Nama menu dan harga harus diisi.

  DPPL:
  1. Terima input dari pengguna.
  2. Validasi input.
  3. Simpan ke array 'menu'.
}

program InsertMenu;

type
  Menu = record
    Name: string;
    Price: real;
  end;

var
  Menus: array[1..100] of Menu;
  Count: integer;

begin
  Count := 0;

  writeln('Masukkan Nama Menu:');
  readln(Menus[Count].Name);
  writeln('Masukkan Harga Menu:');
  readln(Menus[Count].Price);

  if (Menus[Count].Name <> '') and (Menus[Count].Price > 0) then
  begin
    writeln('Menu berhasil ditambahkan!');
    Count := Count + 1;
  end
  else
    writeln('Nama dan Harga tidak boleh kosong.');
end.

